rust/arrow/src/array/array.rs

```diff
@@ -326,6 +327,170 @@ pub fn new_empty_array(data_type: &DataType) -> ArrayRef {
     let data = ArrayData::new_empty(data_type);
     make_array(Arc::new(data))
 }
+/// Creates a new array of `data_type` of length `length` filled entirely of `NULL` values
+pub fn new_null_array(data_type: &DataType, length: usize) -> ArrayRef {
```

Comment:
I am not sure why this code is in this PR (maybe the PR needs a rebase)? It was merged into master as part of https://github.com/apache/arrow/pull/9469 I think